### PR TITLE
Remove use of deprecated RulesDefinitionXmlLoader

### LINKED ISSUE #5590: Remove use of deprecated RulesDefinitionXmlLoader
We generate `rules.xml` file with rule descriptions that is imported by `RulesDefinitionXmlLoader` on the SQ plugin side.

The only useful information that the file contains is a list of parameters of the rules. The rest can be read directly from the JSON files.

We should ~~simplify `rules.xml` to contain only that. And~~ consider moving to json instead.

We should not depend on `RulesDefinitionXmlLoader`, as it is deprecated by SQ, see [SONAR-14908](https://jira.sonarsource.com/browse/SONAR-14908).

**Data flow before:**
```
rspec files -> rspec script -> Analyzer resx --+-> Utilities -> RuleDescriptorGenerator -> rules.xml -> Java plugin
Rule parameters from reflection --------------´
```
**Data flow after:**
```
rspec files ---------------------------------------------------------------------+-> Java plugin
Rule parameters from reflection -> RuleDescriptorGenerator -> parameters.json --´
```

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #5590
Prerequisite for #5210 - this fixes the Java side

There's a lot of unused stuff around `rules.xml` file. Cleanup will follow in other PRs

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs [hunk 1]
```diff
@@ -31,6 +31,7 @@
 using System.Xml.Serialization;
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
+using SonarAnalyzer.Rules;
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs:34]: There are 3 usings that are only needed for NETCOREAPP. Please move them inside the conditional part.

File: sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java [hunk 2]
```diff
@@ -104,14 +114,103 @@ public void test_security_standards_before_9_3() {
...
+    CSharpSonarRulesDefinition definition = new CSharpSonarRulesDefinition(SONAR_RUNTIME);
+    RulesDefinition.Context context = new RulesDefinition.Context();
+    definition.define(context);
+    RulesDefinition.Rule rule = context.repository("csharpsquid").rule(MULTI_PARAM_RULE_KEY);
+
+    assertThat(rule.params()).hasSize(2);
+    assertParam(rule.params().get(0), "max", RuleParamType.INTEGER, "5", "Maximum depth of the inheritance tree. (Number)");
+    assertParam(rule.params().get(1), "filteredClasses", RuleParamType.STRING, null,
+      "Comma-separated list of classes or records to be filtered out of the count of inheritance. Depth counting will stop when a filtered class or record is reached. For example: System.Windows.Controls.UserControl, System.Windows.*. (String)");
+
```
Comment (csaba-sagi-sonarsource) [inline sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java:207]: Extra empty line.

File: analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs:34]: I'll delete most of this file in the next PR. I won't fix it here

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**